Sources/NIO/Bootstrap.swift

```diff
@@ -163,6 +163,41 @@ public final class ServerBootstrap {
         }
     }
 
+    /// Use the existing bound socket file descriptor.
+    ///
+    /// - parameters:
+    ///     - descriptor: The _Unix file descriptor_ representing the bound stream socket.
+    public func withBoundSocket(descriptor: Int32) -> EventLoopFuture<Channel> {
+        let childEventLoopGroup = self.childGroup
+        let serverChannelOptions = self.serverChannelOptions
+        let eventLoop = group.next()
+        let serverChannelInit = self.serverChannelInit ?? { _ in eventLoop.newSucceededFuture(result: ()) }
+        let childChannelInit = self.childChannelInit
+        let childChannelOptions = self.childChannelOptions
+
+        let promise: EventLoopPromise<Channel> = eventLoop.newPromise()
+        do {
+            let serverChannel = try ServerSocketChannel(descriptor: descriptor, eventLoop: eventLoop as! SelectableEventLoop, group: childEventLoopGroup)
+
+            serverChannelInit(serverChannel).then {
+                serverChannel.pipeline.add(handler: AcceptHandler(childChannelInitializer: childChannelInit,
+                                                                  childChannelOptions: childChannelOptions))
+            }.then {
+                serverChannelOptions.applyAll(channel: serverChannel)
+            }.then {
+                let promise: EventLoopPromise<Void> = eventLoop.newPromise()
+                serverChannel.registerAlreadyConfigured0(promise: promise)
+                return promise.futureResult
```

Comment:
This code looks very similar to existing code. Can we share some of it ?

Reply:
Could you hint as to which direction you'd like this code to go? Should I search for an existing function, or just extract this pattern to somewhere? How would you solve that?